conf.py

```diff
@@ -352,8 +352,14 @@ def xapian_run_example_command(ex):
         return "%s -unsafe -target:exe -out:%s.exe %s -r:XapianSharp.dll\n./%s.exe" \
             % (csc, ex, xapian_code_example_filename(ex), ex)
     elif highlight_language == 'java':
-        javac = get_tool_name('JAVAC', 'javac')
-        java = get_tool_name('JAVA', 'java')
+        java_bindings_dir = os.environ.get("JAVA_BINDINGS_DIR")
+        classpath = ''
+        java_library_path = ''
+        if java_bindings_dir is not None:
+            classpath = ' -classpath ' + java_bindings_dir + 'xapian_jni.jar:./code/java/'
```

Comment:
It's important to understand what a classpath string actually does. The colon (`:`) is a path separator (which should perhaps be specified as `os.pathsep` now I think about it — you'd have to consult the Java Windows documentation to be sure). So the things either side of it should be paths. `./code/java/` is a path, but you shouldn't concatenate strings to make the other path, you should use `os.path.join()`.